pkg/svc/atlas_patch.go

```diff
@@ -207,21 +226,43 @@ The service-defined string used to identify a page of resources. A null value in
 			}
 			op.Parameters = fixedParams
 
+			customResponseCode := responseCodesMap[on]
+
 			// Wrap responses
 			if op.Responses.StatusCodeResponses != nil {
 				// check if StatusCodeResponses has 201 >= x < 300 then delete 200 and don't go to isNilRef check
 				exists := false
-				for code := range op.Responses.StatusCodeResponses {
-					if code >= 201 && code < 300 {
-						exists = true
+				if customResponseCode != 200 {
+					for code := range op.Responses.StatusCodeResponses {
+						if code >= 201 && code < 300 {
+							exists = true
+						}
+						break
 					}
-					break
 				}
+
+				defaultResponseCode := 200
+				if customResponseCode == 200 {
+					defaultResponseCode = DefaultResponseCodesMap[on]
+				}
+
 				if exists {
-					delete(op.Responses.StatusCodeResponses, 200)
+					delete(op.Responses.StatusCodeResponses, defaultResponseCode)
```

Comment:
Is it necessary to change `200` to `defaultResponseCode` here? I think if you are in this block, you always have `defaultResponseCode = 200`.

Reply:
No, the default response code is different for the different API calls, like 201 for Post, 200 for Get.

Reply:
`defaultResponseCode` is set to `200` on line 235. The only line it can change is line 237. Line 237 hasn't be executed in case `exists == true`.

Reply:
It is probably okay to leave 200 for this line.  Will update.